### DISCUSSION TITLE
Date sort on managed property is giving HTTP:500 Error and modern search web part is not working

### DISCUSSION THREAD

**Discussion by RK9080** [microsoft-search/pnp-modern-search#2603] [Q&A]
### Issues related to PnP Search Results Web Part - Issues related to date column sorting

**Version used**
PnP Search Results Web Part V4
Version: 4.7.0.0

**Describe the bug**
Search Results web part breaks on date column sorting and gives HTTP:500 error. Date Column is kept sortable in the manage columns. 

**To Reproduce**
1. Use date column in the PnP search results web part with details list view layout
2. Click on date column for sorting on asc/desc order
3. Web part does not work for sorting and gives error
4. See error - [HTTP]:500 - [CorrelationId]:cc7a7ba0-a054-2000-d7ce-aa35ed86b9a6 [Version]:16.0.0.23102

**Expected behavior**
Web Part should sort the data once clicked on the date column header for sorting in asce/desc order

**Screenshots**
![image](https://user-images.githubusercontent.com/118924924/203612088-53d7afac-58cc-476f-bbf1-d36dea5d4dc0.png)

**Desktop (please complete the following information):**
 - Browser [e.g. chrome, edge]

**Additional context**
Add any other context about the problem here - Date column is regular column in the SPO document library

**Comment by wobba** [comment]
Which managed property are you sorting on, and have you checked that it is sortable/refinable? 500 errors are usually an invalid query somehow. You also don't mention which search provider you are using. Repro without detailed information is difficult.

**Comment by RK9080** [comment]
Thanks for your reply.

In this case managed property is "Published Date"  (SPO document library column ) column shown in the screenshot. By default for this column's managed property in the search schema - sortable value is "No" and grayed out and can not be changed to "Yes".  
Search Provider :  SharePoint Search &
Result Source: Local SharePoint Results

Search Web Part breaks only on sorting for date column

PnP web part crashes in the core js when clicked on the "Published Date" column for sorting in asc/desc . Here is the debug trace

![image](https://user-images.githubusercontent.com/118924924/204331437-ee11b611-3a79-4b53-94f2-27d50fb4bebc.png)

 
Let me know if you have any suggestions to fix this issue. Thanks.

**Comment by wobba** [comment]
You need to map the crawled property for PublishedDate to a RefinableDate property and use that one instead. As refinable props are also sortable.

**Reply by RK9080** [reply]
Yes. First time I did the same to fix this issue but data is not coming for the newly created managed property.

the default "Published Date" column is site column for which managed property is created which is used in the Search web part and data is getting crawled but sorting does not works. I also tried to add this column in the sort settings column to apply sort but after adding this column then it starts giving http:500 error all the time

**Reply by wobba** [reply]
Automatic managed properties from site columns are all of type text, and not sortable/refinable. So you have to do manual mapping of crawled properties to one of the reusable managed properties to make this work.
